develop/app/Controllers/v2/PaymentController.php

```diff
@@ -0,0 +1,262 @@
+
+        $data = [
+            "list"      => [],
+            "dataCount" => $dataCount
+        ];
+
+        if ($payments) {
+            foreach ($payments as $paymentEntity) {
+                $paymentData = [
+                    "pm_key" => $paymentEntity->pm_key,
+                    "u_key"  => $paymentEntity->u_key,
+                    "o_key"  => $paymentEntity->o_key,
+                    "status" => $paymentEntity->status,
+                    "total"  => $paymentEntity->total
+                ];
+                $data["list"][] = $paymentData;
+            }
+        } else {
+            return $this->fail("Payment data not found", 404);
+        }
+
+        return $this->respond([
+            "status" => true,
+            "data"   => $data,
+            "msg"    => "Payment index method successful",
+        ]);
+    }
+
+    /**
+     * [GET] /api/v1/payment/{paymentKey}
+     * Get payment data by payment key.
+     *
+     * @param int $paymentKey
+     * @return void
+     */
+    public function show($paymentKey = null)
+    {
+        if ($paymentKey === null) {
+            return $this->fail("The payment key is required.", 404);
+        }
+
+        $paymentModel = new PaymentModel();
+
+        $paymentEntity = $this->getPaymentEntityByPaymentKey($paymentKey);
+        if (is_null($paymentEntity)) {
+            return $this->fail("This payment information is not exist or cannot found.", 404);
+        }
+
+        $data = [
+            "pm_key" => $paymentEntity->pm_key,
+            "u_key"  => $paymentEntity->u_key,
+            "o_key"  => $paymentEntity->o_key,
+            "status" => $paymentEntity->status,
+            "total"  => $paymentEntity->total
+        ];
+
+        return $this->respond([
+            "status" => true,
+            "data"   => $data,
+            "msg"    => "Payment show method successful",
+        ]);
+    }
+
+    /**
+     * [POST] /api/v1/payment
+     * Create payment and reduce user balance.
+     *
+     * @return void
+     */
+    public function create()
+    {
+        $data   = $this->request->getJSON(true);
+        $u_key  = $this->u_key;
+        $o_key  = $data["o_key"];
+        $amount = $data["amount"] ?? null;
+        $price  = $data["price"]  ?? null;
+        $status = "paymentCreate";
+
+        if (is_null($u_key) || is_null($o_key) || is_null($amount) || is_null($price)) {
+            return $this->fail("Incoming data not true", 400);
+        }
+
+        $total = $amount * $price;
+
+        $paymentModel  = new PaymentModel();
+        $walletModel   = new WalletModel();
+
+        $paymentEntity = $paymentModel->where("u_key", $u_key)
+                                      ->where("o_key", $o_key)
+                                      ->first();
+        if (!is_null($paymentEntity)) {
+            return $this->fail("This payment information is not exist or cannot found.", 403);
```

Comment:
is not information` and `cannot found` mean the same thing, doesn't it?

Reply:
This change is done.